_posts/2021-03-10-application-modernisation-101.markdown

```diff
@@ -0,0 +1,74 @@
+
+## Replace/Repurchase
+So it turns out that your unicorn application isn’t so niche after all, and there’s a SaaS provider that provides the same services. Why are you maintaining the application at all, when that effort could be put into core business? Perhaps simply purchasing the services from a SaaS provider is the best way forward. Just make sure you don’t lose needed capabilities and consider the consequences of potential lock in. 
+
+## Rehost
+Yes.. it’s lift and shift. Yes, it’s quick and dirty. This isn’t really application modernisation at all, it’s merely moving your applications from one place to another. It’s a super quick win where you can claim your applications have moved to the cloud, but you aren’t really going to see any of the benefits. You still need to manage the operating system patches and your own scalability and availability. Only consider this option if you’re considering sunset applications that will be obsolete or retired in a few years, or your data centre is closing down tomorrow and you need a stop-gap measure.
+
+## Replatform
+Replatforming your application requires strategic and selective migration of services to managed services offerings. This means undertaking tasks such as migration from databases running on on-prem servers to managed services like RDS and applications to container-based infrastructure. This is a solid approach that minimises application redevelopment and still started to benefit from the scalability, redundancy and fault tolerance capabilities of cloud based infrastructure. It does require more effort ($$$) than simple rehosting, however the total cost of ownership savings are significant.
+
+## Refactor
+There’s a couple of ways to tackle refactoring your application. This can range from adopting the Strangler Pattern and methodically replacing and rewriting parts of your application, through to complete wholesale rewrites. The aim with refactoring your application, will be to leverage Platform as a Service components of your chosen cloud provider to get the best resilience, scalability and performance. Refactoring your application to use serverless technologies will see great improvements in its efficiency, where you’re only paying for what you use, and are able to scale up effortlessly. Critics of this Cloud Native will often quote vendor lock-in as the major issue with this approach, and this is true - you will be tied to the platform provider’s implementation of particular services, however… this is also true of any software development project using any framework, whether cloud native or not. As with replatforming, refactoring requires significant effort and will often have the largest outlays of any of the application modernisation strategies, however it also provides the biggest cost savings in the long term, and positions the application in the best position for future maintenance, enhancement and scaling.
```

Comment:
Could we use the SmartApps stats that Tim just got for Dept of Communities?

Reply:
yep